.github/workflows/test_develop_cli.yaml

```diff
@@ -0,0 +1,79 @@
+name: test-develop-cli
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches:
+      - develop
+      - 'pre/*' # We should reduce the frequency of these builds to save resources since it tests the docs build which is pretty long.
```

Comment:
So I was thinking just triggered on the `develop`  PR or manually, how does that sound?

The nice thing about this script is that it tests that the docs build, unsure how to test for a release since it is just creating a tag that triggers that? Could add a pre trigger run if we wanted on any release?

Reply:
seems like 80% of the time we're developing on a pre/* branch, so probably we should be running on those.

What if we tested after a new commit is added to the pre/* branch (so AFTER a PR is merged?) or is this looking for trouble?

I think on release it might be too late.. we'd just have to patch a new release right after if it breaks.

Reply:
So the tricky thing is these develop tests will also test the documentation build which is several minutes long. If we don't mind that then I guess it'd be fine to run on pre/ anytime